components/WS/src/HttpServerHandler.h

```diff
@@ -0,0 +1,231 @@
+        pHttpResponse->setStatus(HttpResponse::HTTP_STATUS_OK, "OK");
+        pHttpResponse->addHeader(HttpRequest::HTTP_HEADER_CONTENT_TYPE, "application/json");
+
+        s1 << "{";
+
+        s1 << "\"voltage\":\"";
+        s1 << voltage;
+        s1 << "\",";
+
+        s1 << "\"avg_current\":\"";
+        s1 << avg_current;
+        s1 << "\",";
+
+        s1 << "\"temp\":\"";
+        s1 << temp;
+        s1 << "\",";
+
+        s1 << "\"avg_power\":\"";
+        s1 << avg_power;
+        s1 << "\",";
+
+        s1 << "\"free_memory\":\"";
+        s1 << heap_caps_get_free_size(MALLOC_CAP_DEFAULT);
+        s1 << "\"";
+
+        s1 << "}";
+
+        pHttpResponse->sendData(s1.str());
+        pHttpResponse->close();
+    }
+
+    /*
+    
+    */
+    static void setUpWstApHandler(HttpRequest* pHttpRequest, HttpResponse* pHttpResponse)
+    {
+        std::string body = pHttpRequest->getBody();
+        std::istringstream ss(body);
+        std::string token[3]; //length of json object, hoe many field there are
+        std::map<std::string, std::string> json_map;
+        int i = 0;
+        size_t found;
+
+        while (std::getline(ss, token[i], ',')) {
+            //to remove open brace in json string packet
+            if (i == 0) {
+                found = token[i].find('{');
+                if (found != std::string::npos) {
+                    token[i].erase(std::remove(token[i].begin(), token[i].end(), '{'), token[i].end());
+                }
+            }
+
+            size_t length = sizeof(token) / sizeof(*token);
+            //to remove clode brace in json string packet
+            if (i == length - 1) {
+                found = token[i].find('}');
+                if (found != std::string::npos) {
+                    token[i].erase(std::remove(token[i].begin(), token[i].end(), '}'), token[i].end());
+                }
+            }
+            //to remove double quote in key value pair
+            found = token[i].find('\"');
+            if (found != std::string::npos) {
+                token[i].erase(std::remove(token[i].begin(), token[i].end(), '\"'), token[i].end());
+            }
+            //to remove tabs in key and value pair
+            found = token[i].find('\t');
+            if (found != std::string::npos) {
+                token[i].erase(std::remove(token[i].begin(), token[i].end(), '\t'), token[i].end());
+            }
+            //to remove new line in key value pair
+            found = token[i].find('\n');
+            if (found != std::string::npos) {
+                token[i].erase(std::remove(token[i].begin(), token[i].end(), '\n'), token[i].end());
+            }
+
+            std::istringstream stream(token[i]);
+            std::string field[2];
+            int j = 0;
+
+            while (std::getline(stream, field[j], ':')) {
+                j++;
+            }
+            json_map.insert(std::make_pair(field[0], field[1]));
+
+            i++;
+        }
```

Comment:
What if we use [cJSON](https://github.com/DaveGamble/cJSON) for this? It isn't that big